sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -341,9 +419,17 @@ export class ManagementClient extends LinkEntity {
    * @param {number} [messageCount] The number of messages to retrieve. Default value `1`.
    * @returns Promise<ReceivedSBMessage[]>
    */
-  async peek(messageCount?: number): Promise<ReceivedMessage[]> {
+  async peek(
+    messageCount?: number,
+    receiverOptions?: GetReceiverOptions
```

Comment:
The operations on the $management link are agnostic of whether a "receiver" is in use or a "sender". I'd expect the input parameter here to be `retryOptions` instead of `receiverOptions`

Reply:
Actually, I've moved to a new design(more commits from Friday - pushed after your review)..

> async peek(messageCount?: number, timeoutInMs?: number)

For the operations under `ManagementClient`, I'm now passing the `timeout` param only and moved the retry logic to a higher level.. 
to handle the retries for the additional `createMessageSessionIfDoesntExists` call that was awaited for each of the methods for sessions(`sessionReceiver.ts`).
Accordingly, updated the receiver methods and subscriptionRuleManager methods.